src/protocols/secure_channel/CASESession.cpp

```diff
@@ -284,30 +279,51 @@ void CASESession::OnResponseTimeout(ExchangeContext * ec)
 
 CHIP_ERROR CASESession::DeriveSecureSession(CryptoContext & session) const
 {
-    size_t saltlen;
-
-    (void) kKDFSEInfo;
-    (void) kKDFSEInfoLength;
+    switch (mState)
+    {
+    case State::kFinished: {
+        // Generate Salt for Encryption keys
+        size_t saltlen = sizeof(mIPK) + kSHA256_Hash_Length;
 
-    VerifyOrReturnError(mCASESessionEstablished, CHIP_ERROR_INCORRECT_STATE);
+        chip::Platform::ScopedMemoryBuffer<uint8_t> msg_salt;
+        ReturnErrorCodeIf(!msg_salt.Alloc(saltlen), CHIP_ERROR_NO_MEMORY);
+        {
+            Encoding::LittleEndian::BufferWriter bbuf(msg_salt.Get(), saltlen);
+            bbuf.Put(mIPK, sizeof(mIPK));
+            bbuf.Put(mMessageDigest, sizeof(mMessageDigest));
 
-    // Generate Salt for Encryption keys
-    saltlen = sizeof(mIPK) + kSHA256_Hash_Length;
+            VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        }
 
-    chip::Platform::ScopedMemoryBuffer<uint8_t> msg_salt;
-    ReturnErrorCodeIf(!msg_salt.Alloc(saltlen), CHIP_ERROR_NO_MEMORY);
-    {
-        Encoding::LittleEndian::BufferWriter bbuf(msg_salt.Get(), saltlen);
-        bbuf.Put(mIPK, sizeof(mIPK));
-        bbuf.Put(mMessageDigest, sizeof(mMessageDigest));
+        ReturnErrorOnFailure(session.InitFromSecret(ByteSpan(mSharedSecret, mSharedSecret.Length()),
+                                                    ByteSpan(msg_salt.Get(), saltlen),
+                                                    CryptoContext::SessionInfoType::kSessionEstablishment, mRole));
 
-        VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        return CHIP_NO_ERROR;
     }
+    case State::kFinishedResumed: {
+        // Generate Salt for Encryption keys
+        size_t saltlen = sizeof(mInitiatorRandom) + mResumeResumptionId.size();
+
+        chip::Platform::ScopedMemoryBuffer<uint8_t> msg_salt;
+        ReturnErrorCodeIf(!msg_salt.Alloc(saltlen), CHIP_ERROR_NO_MEMORY);
+        {
+            Encoding::LittleEndian::BufferWriter bbuf(msg_salt.Get(), saltlen);
+            bbuf.Put(mInitiatorRandom, sizeof(mInitiatorRandom));
+            bbuf.Put(mResumeResumptionId.data(), mResumeResumptionId.size());
+
+            VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        }
 
-    ReturnErrorOnFailure(session.InitFromSecret(ByteSpan(mSharedSecret, mSharedSecret.Length()), ByteSpan(msg_salt.Get(), saltlen),
-                                                CryptoContext::SessionInfoType::kSessionEstablishment, mRole));
+        ReturnErrorOnFailure(session.InitFromSecret(ByteSpan(mSharedSecret, mSharedSecret.Length()),
+                                                    ByteSpan(msg_salt.Get(), saltlen),
+                                                    CryptoContext::SessionInfoType::kSessionResumption, mRole));
```

Comment:
It looks like the callee does not handle kSessionResumption per spec.  It uses an "info" value that does not match the spec.  Followup to fix that, please?

Reply:
Fixed in this PR